ksqldb-parser/src/main/java/io/confluent/ksql/parser/tree/CreateTable.java

```diff
@@ -106,13 +107,22 @@ private static void throwOnNonPrimaryKeys(final TableElements elements) {
 
     wrongKey.ifPresent(col -> {
       final String loc = NodeLocation.asPrefix(col.getLocation());
+      final String fullMessage = loc + "Column " + col.getName() + " is a 'KEY' column: "
```

Comment:
As an idle comment, I wish it was easy in Java to pass around lambdas.  

If we are just throwing the exception, it'd be neat to avoid computing the full message.

(He says over-optimizing.)

Reply:
It's pretty easy nowadays. I was just more focused on getting everything working than making it nice.